integration-tests/src/test/java/oracle/weblogic/kubernetes/ItManageNs.java

```diff
@@ -0,0 +1,614 @@
+   * Add label1 to ns1 and verify domain can't be started
+   * Call upgrade operator with reuse values to enable management for ns1
+   * Deploy a custom domain resource in the namespace ns1 with label1
+   * and verify all server pods in the domain were created and ready.
+   * Verify operator is able to manage this domain by scaling.
+   */
+  @Test
+  @Order(3)
+  @DisplayName("install operator helm chart and domain, "
+      + " with enableClusterRoleBinding")
+  public void testSwitchRbac() {
+    String manageByLabelDomainNS = domainNamespaces[0] + "test4";
+    String manageByLabelDomainUid = domainsUid[0] + "test4";
+    assertDoesNotThrow(() -> createNamespace(manageByLabelDomainNS));
+    opHelmParams[2] = installAndVerifyOperator(OPERATOR_RELEASE_NAME,
+        opNamespaces[3], "LabelSelector",
+        "mytest4", false);
+    Map<String, String> labels = new HashMap<>();
+    labels.put("mytest4", manageByLabelDomainUid);
+    assertDoesNotThrow(() -> addLabelsToNamespace(manageByLabelDomainNS, labels));
+    //verify domain can't be started because operator does not have permission to manage it
+    createSecrets(manageByLabelDomainNS);
+    checkPodNotCreated(manageByLabelDomainUid + adminServerPrefix, manageByLabelDomainUid, manageByLabelDomainNS);
+    deleteDomainCrd(manageByLabelDomainNS, manageByLabelDomainUid);
+    //upgrade operator and start domain
+    int externalRestHttpsPort = getServiceNodePort(opNamespaces[3], "external-weblogic-operator-svc");
+
+    OperatorParams opParams = new OperatorParams()
+        .externalRestEnabled(true)
+        .externalRestHttpsPort(externalRestHttpsPort)
+        .helmParams(opHelmParams[2]);
+
+    assertTrue(upgradeAndVerifyOperator(opNamespaces[3], opParams));
+    assertTrue(createDomainCrdAndVerifyDomainIsRunning(manageByLabelDomainNS, manageByLabelDomainUid));
+    checkOperatorCanScaleDomain(opNamespaces[3], manageByLabelDomainUid);
+  }
+
+  private void checkUpgradeFailedToAddNSManagedByAnotherOperator() {
+    //upgrade operator1 to replace managing domains using RegExp namespaces
+    // for ns names starting from weblogic, there one of domains
+    //in namespace weblogic* is managed by operator2
+    int externalRestHttpsPort = getServiceNodePort(opNamespaces[0], "external-weblogic-operator-svc");
+    //set helm params to use domainNamespaceSelectionStrategy=RegExp for namespaces names started with weblogic
+    OperatorParams opParams = new OperatorParams()
+        .helmParams(opHelmParams[0])
+        .externalRestEnabled(true)
+        .externalRestHttpsPort(externalRestHttpsPort)
+        .domainNamespaceSelectionStrategy("RegExp")
+        .domainNamespaceRegExp("^" + "weblogic");
+
+    assertFalse(upgradeAndVerifyOperator(opNamespaces[0], opParams), "Upgrade does not fail when adding domain,"
+        + " managed by other operator");
+  }
+
+  private void deleteDomainCrd(String domainNS, String domainUid) {
```

Comment:
Here we are not deleting Domain CRD which is a global resource created by Operator. Actually we are deleting Domain Resource ( domain ).  Modify the method name to  deleteDomainResource()

Reply:
done